include/api.php

```diff
@@ -5098,15 +5098,18 @@ function api_friendica_remoteauth()
 	// traditional DFRN
 
 	$contact = DBA::selectFirst('contact', [], ['uid' => api_user(), 'nurl' => $c_url]);
-
-	if (!DBA::isResult($contact) || ($contact['network'] !== Protocol::DFRN)) {
+	if (!DBA::isResult($contact)) {
 		throw new BadRequestException("Unknown contact");
 	}
 
 	$cid = $contact['id'];
 
 	$dfrn_id = $contact['issued-id'] ?? $contact['dfrn-id'];
 
+	if (($contact['network'] !== Protocol::DFRN) || empty($dfrn_id)) {
+		System::externalRedirect($url ?? $c_url);
```

Comment:
`$url` will never be `null` since it's initialized with empty string if `$_GET['url']` doesn't exist. As a consequence this will always redirect to `$url`. You should use `?:` instead of `??`.
```suggestion
		System::externalRedirect($url ?: $c_url);
```

Reply:
Ah okay. Then shouldn't the line with the "dfrn_id" be changes as well? This won't be null, but be an empty string.

Reply:
Absolutely, good catch.